corehq/apps/user_importer/validation.py

```diff
@@ -136,12 +138,31 @@ def validate_spec(self, spec):
 class TableauRoleValidator(ImportValidator):
     _error_message = _("Invalid tableau role: '{}'. Please choose one of the following: {}")
 
+    def __init__(self, domain):
+        super().__init__(domain)
+        self.valid_role_options = [option[1] for option in TABLEAU_ROLES]
+
     def validate_spec(self, spec):
-        valid_role_options = [option[1] for option in TABLEAU_ROLES]
         tableau_role = spec.get('tableau_role')
+        if tableau_role is not None and tableau_role not in self.valid_role_options:
+            return self._error_message.format(tableau_role, ', '.join(self.valid_role_options))
+
+
+class TableauGroupsValidator(ImportValidator):
+    _error_message = _("These groups, {}, are not valid for this domain. Please choose one of the following: {}")
```

Comment:
very very nit: would "Please choose one of the following" give the impression that one and only one group can be chosen? Maybe something like "Please choose from the following:"

Reply:
yes, good catch